lib/linux/sys/proctable.rb

```diff
@@ -169,6 +171,10 @@ def self.ps(pid=nil)
         # Get control groups to which the process belongs
         struct.cgroup = IO.readlines("/proc/#{file}/cgroup").map { |l| CgroupEntry.new(l) } rescue []
 
+        # Read smaps, return a parseable empty string if we don't have permissions
+        smaps_contents = IO.read("/proc/#{file}/smaps") rescue ""
```

Comment:
From our chat... proc filesystem is weird and ruby's File class can't tell we're not allowed to read files, I'll add a comment for this part:

```ruby
irb(main):001:0> File.readable?('/proc/1/smaps')
=> true
irb(main):002:0> File.readable_real?('/proc/1/smaps')
=> true
irb(main):003:0> File.read('/proc/1/smaps')
Errno::EACCES: Permission denied - /proc/1/smaps
```